Jenkinsfile

```diff
@@ -89,8 +89,6 @@ MAVEN_PROFILES=${env.MAVEN_PROFILES}
                                 -B \
                                 $MAVEN_PROFILES \
                                 -Dmaven.repo.local=$WORKSPACE/.m2/repository \
-                                -Dmaven.test.failure.ignore=true \
```

Comment:
Is it really related with the rest of changes?

Reply:
Yes, this is important, as we don't want to cache failing tests, which is caused by using this flag. I can also open a separate PR just with this, but it is related. 

An alternative that works well with caching would be to use the --fail-at-end flag instead, to run all the tests (except for the ones downstream from the module containing the test failure), but still fail the build at the end. 

Note: I'm happy to squash the commits, but I suggest we resolve any open questions beforehand.

Reply:
ok, let's keep it here and give it a try